cardano-db/src/Cardano/Db/Migration/Extra/CosnumedTxOut/Queries.hs

```diff
@@ -202,40 +206,70 @@ countConsumed = do
     pure countRows
   pure $ maybe 0 unValue (listToMaybe res)
 
+deleteAndUpdateConsumedTxOut :: MonadIO m => Trace IO Text -> Word64 -> ReaderT SqlBackend m ()
+deleteAndUpdateConsumedTxOut trce blockNoDiff = do
+  maxTxInId <- findMaxTxInId blockNoDiff
+  case maxTxInId of
+    Left errMsg -> liftIO $ logInfo trce $ "No tx_out were deleted: " <> errMsg
+    Right mxtxid -> do
+      migrateNextPage mxtxid 0
+  where
+    migrateNextPage :: MonadIO m => TxInId -> Word64 -> ReaderT SqlBackend m ()
+    migrateNextPage mxTxInId offst = do
+      page <- getInputPage offst pageSize
+      foldM_ (handlePageEntry mxTxInId) False page
+      when (fromIntegral (length page) == pageSize) $
+        migrateNextPage mxTxInId $!
+          offst
+            + pageSize
+
+    handlePageEntry :: MonadIO m => TxInId -> Bool -> (TxInId, TxId, Word64) -> ReaderT SqlBackend m Bool
+    handlePageEntry maxTxInId ranCreateConsumed (txInId, txId, index)
+      | txInId <= maxTxInId = do
+          deleteTxOutConsumed txId index
+          pure False
+      | otherwise = do
+          -- we want to call createConsumedTxOut just once
+          unless ranCreateConsumed createConsumedTxOut
```

Comment:
@kderme A problem might be this per "page" so each time a new page is called it will reset `ranCreateConsumed` to `False` might have to change `migrateNextPage` to be a `FoldM` too and return `(ReaderT SqlBackend m (Int, Bool)` 🤔